src/containers/pages/IRS/assignments/index.tsx

```diff
@@ -99,7 +98,7 @@ const IRSAssignmentPlansList = (props: PlanAssignmentsListProps) => {
     loadData().catch(error => displayError(error));
   }, []);
 
-  if (loading) {
+  if (plans.length < 1) {
```

Comment:
If we get 0 plans from the API, does this mean we will show the loading indicator forever?

Reply:
unfortunately yes, the alternative is to start implementing the changes that arise from the discussions here https://github.com/onaio/reveal-frontend/issues/806. This change only attempts to stop with setting state error, but still retains the same logic for when the loader is going to be displayed.

Reply:
i was thinking about this wrong. let me refactor.